### PR TITLE
perf: small queries to identify a devHub

### PR DESCRIPTION
**Pull request by mshanemc**
related to work for @W-8881661@

there's no reason to query every scratchOrgInfo (on a busy hub that someone's authenticating a new machine to, that could be a lot of data.  

The sobject either exists or doesn't so this'll be faster.

### CODE REVIEW

File: src/org.ts [hunk 1]
```diff
@@ -233,7 +233,7 @@ export class Org extends AsyncCreatable<Org.Options> {
     const conn = this.getConnection();
     let isDevHub = false;
     try {
-      await conn.query('SELECT Id FROM ScratchOrgInfo');
+      await conn.query('SELECT Id FROM ScratchOrgInfo limit 1');
```
Comment (peternhale) [inline src/org.ts:236]: What's cheaper a soql query or sobject describe?

**Review by peternhale — APPROVED** [review]

**Review by amphro — APPROVED** [review]

File: src/org.ts (hunk 1 above)
Comment (WillieRuemmele) [reply src/org.ts:236]: couldn't you use `singleRecordQuery`? If the query returning 2+ results is a bug we should have a test to verify we don't break this in the future

Comment (mshanemc) [reply src/org.ts:236]: query

Comment (mshanemc) [reply src/org.ts:236]: we don't care how many come back.  0 records is fine for a virgin devhub...
we only care that the obj exists at all and doesn't throw an error

Comment (mshanemc) [reply src/org.ts:236]: I'm setting it to `1` so busy hubs don't have to return 50000 or whatever.

**Review by shetzel — APPROVED** [review]

**Pull request merged**